<launch>

    <node pkg="lio_livox" type="ScanRegistration" name="ScanRegistration" output="screen">

        <param name="config_file" value="$(find lio_livox)/config/mid360_config.yaml"/>
        <!--0-custom msg ,1 ros sensor msg pointcloud2 msg  -->
        <param name="msg_type" type="int" value="1"/>
    </node>

    <node pkg="lio_livox" type="PoseEstimation" name="PoseEstimation" output="screen">
        <!-- 0-Not Use IMU, 1-Use IMU remove Rotation Distort, 2-Tightly Coupled IMU -->
        <param name="IMU_Mode" type="int" value="2" />
        <!-- Voxel Filter Size Use to Downsize Map Cloud -->
        <param name="filter_parameter_corner" type="double" value="0.2" />
        <param name="filter_parameter_surf" type="double" value="0.4" />
        <!-- Extrinsic Parameter between Lidar & IMU -->
        <rosparam param="Extrinsic_Tlb"> [0.9999161, 0.0026676,  0.0126707, -0.011,
                                         -0.0025826, 0.9999741, -0.0067201, -0.0234,
                                         -0.0126883, 0.0066868,  0.9998971,   0.044,
                                                0.0,       0.0,        0.0,      1.0]</rosparam>
    </node>

    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find lio_livox)/rviz_cfg/lio.rviz" />

</launch>
